model_signing/signature/signing.py

```diff
@@ -0,0 +1,209 @@
+        subjects = stmnt.pb.subject
+        sigstore_subjects = []
+        for s in subjects:
+            sigstore_subjects.append(
+                dsse._Subject(
+                    name=s.name,
+                    digest={"sha256": s.digest["sha256"]},
+                )
+            )
+        return dsse._StatementBuilder(
+            predicate_type=stmnt.pb.predicate_type,
+            predicate=stmnt.pb.predicate,
+            subjects=sigstore_subjects,
+        ).build()
+
+    @staticmethod
+    def __get_identity_token(
+        disable_ambient: bool = True,
+        id_provider: Optional[str] = None,
+    ) -> Optional[oidc.IdentityToken]:
+        token: oidc.IdentityToken
+        if not disable_ambient:
+            return oidc.detect_credential()
+
+        issuer = oidc.Issuer(id_provider) if id_provider else oidc.Issuer.production()
+        token = issuer.identity_token(
+            client_id=SigstoreSigner.CLIENT_ID, force_oob=True
+        )
+        return token
+
+    def sign(self, stmnt: statement.Statement) -> bundle_pb.Bundle:
+        return self._signer.sign_dsse(self.__convert_stmnt(stmnt))._inner
+
+
+class KeySigner(Signer):
+    """Provides a Signer using an elliptic curve private key for signing."""
+
+    def __init__(self, private_key_path: str, password: Optional[str] = None):
+        self._private_key: crypto_ec.EllipticCurvePrivateKey
+        with open(private_key_path, "rb") as fd:
+            serialized_key = fd.read()
+        self._private_key = crypto_serialization.load_pem_private_key(
+            serialized_key, password=password
+        )
+
+    def sign(self, stmnt: statement.Statement) -> bundle_pb.Bundle:
+        pae = utils.pae(stmnt.pb)
+        sig = self._private_key.sign(pae, crypto_ec.ECDSA(SHA256()))
+        env = intoto_pb.Envelope(
+            payload=json_format.MessageToJson(stmnt.pb).encode(),
+            payload_type=utils.PAYLOAD_TYPE,
+            signatures=[intoto_pb.Signature(sig=sig, keyid=None)],
+        )
+        bdl = bundle_pb.Bundle(
+            media_type='application/vnd.dev.sigstore.bundle.v0.3+json',
+            verification_material=bundle_pb.VerificationMaterial(
+                public_key=common_pb.PublicKey(
+                    raw_bytes=self._private_key.public_key().public_bytes(
+                        encoding=crypto_serialization.Encoding.PEM,
+                        format=crypto_serialization.PublicFormat.SubjectPublicKeyInfo,
+                    ),
+                    key_details=common_pb.PublicKeyDetails.PKIX_ECDSA_P256_SHA_256,
+                ),
+            ),
+            dsse_envelope=env,
+        )
+
+        return bdl
+
+
+class PKISigner(Signer):
+    """Provides a Signer using an elliptic curve private key for signing and
+    adds the provided certificate information as verification material."""
+
+    def __init__(
+        self, private_key_path: str, signing_cert_path: str, cert_chain_paths: list[str]
```

Comment:
instead of passing in signing_cert_path and cert_chain_paths as str, shall we pass them as bytes or iterators? This way a caller can call this API without the need to copy certs to a file / folder. We can provide default constructors like `SigningCertsBuilder.from_path()`.

Reply:
I changed it to accept the private key, the signing cert and a list of certs. Also added the from_path mehtod